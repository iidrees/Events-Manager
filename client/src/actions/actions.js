import axios from 'axios';
import { history } from '../routes';

import { 
  SIGN_UP, SIGNED_UP, 
  SIGN_UP_FAIL, SIGN_IN, 
  SIGNED_IN, SIGN_IN_FAIL 
} from './types';

/**
 * Axios will help make POST request to signup user
 * @export {function}
 * @param {any} userData
 * @returns {JSON} userData
 */
export const userSignup = (userData) => {
  return (dispatch) => {
    dispatch({ type: 'SIGN_UP' });
    axios({
      method: 'post',
      url: 'http://localhost:5050/api/v1/users',
      data: userData,
      withCredentials: true,
    })
      .then((response) => {
        dispatch({ type: SIGNED_UP, payload: response.data });
        history.push('/getEvents');
      })
      .catch((err) => {
        dispatch({ type: 'SIGN_UP_FAIL', payload: err });
        history.push('/');
      });
  };
};

export const userSignin = (userData) => {
  return (dispatch) => {
    dispatch({ type: 'SIGN_IN' });
    axios({
      method: 'post',
      url: 'http://localhost:5050/api/v1/users/login',
      data: userData,
      withCredentials: true,
    })
      .then((response) => {
        dispatch({ type: SIGNED_IN, payload: response.data });
        history.push('/login');
      })
      .catch((err) => {
        console.log(err.response.data);
        dispatch({ type: 'SIGNED_IN_FAIL', payload: err });
      // history.push('/login);
      });
  };
};

/**
 * @export {function}
 * @param {any} userData
 * @returns {JSON} userData
 */
/* export function userSigninRequest(userData) {
  return dispatch => axios.post('/api/v1/users/login', userData);
}
 */
