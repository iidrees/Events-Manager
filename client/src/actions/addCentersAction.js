import axios from 'axios';

import {
  ADD_CENTER,
  ADD_CENTER_FAIL
} from './types'

/* eslint-disable */
export const addCenter = (centerData) => {
  console.log('centerData from actions', centerData);
  return (dispatch) => {
    axios({
      method: 'post',
      url: '/api/v1/centers',
      data: centerData,
      headers: {
        'x-access-token': localStorage.getItem('x-access-token')
      },
      withCredentials: true
    })
    .then((res) => {
      console.log('res from actions addcenter',  res);
      dispatch(({ type: ADD_CENTER, res }))
    })
    .catch((err) => {
      console.log('err from actions addcenters', err);
      dispatch({ type: ADD_CENTER_FAIL, err });
    })
  }
}