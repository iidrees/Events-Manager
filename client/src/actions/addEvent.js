/* Import dependencies and Modules */

import axios from 'axios';
import { history } from '../routes';

import {
  ADD_EVENT,
  ADD_EVENT_FAIL,
  ADD_IMG_FAIL,
  GET_EVENT,
  ADD_EVENT_LOAD
} from './types';

/**
 * Axios will help make POST request to add event
 * @export {function} addEvent function action
 * @param {eventData} eventData object sent to the server
 * @param {centerId}  centerId {centerId}
 * @param {imgUrl} imgUrl {the cloudinary url}
 * @returns {JSON} JSON data containing events
 */
export const addEvent = (eventData, centerId, imgUrl) => {
  return dispatch => {
    return axios({
      method: 'post',
      url: `/api/v1/events/${centerId}`,
      data: {
        title: eventData.title,
        startDate: eventData.startDate,
        endDate: eventData.endDate,
        center: eventData.center,
        description: eventData.description,
        imgUrl: imgUrl
      },
      headers: {
        'x-access-token': localStorage.getItem('x-access-token')
      },
      withCredentials: true
    })
      .then(response => {
        dispatch({ type: ADD_EVENT, event: response.data });
      })
      .catch(err => {
        dispatch({ type: ADD_EVENT_FAIL, error: err.response.data });
      });
  };
};

/**
 * @export {function} imageUpload function
 * @param {eventData} eventData object sent to the server
 * @param {centerId}  centerId {centerId}
 * @returns {URL} URL link returned is used as a parameter
 */
export const imageUpload = (eventData, centerId) => {
  let formData = new FormData();

  formData.append('file', eventData.imgFile);
  formData.append('upload_preset', process.env.UPLOAD_PRESET);
  return dispatch => {
    dispatch({ type: ADD_EVENT_LOAD });
    axios({
      method: 'post',
      url: process.env.CLOUDINARY_URL,
      data: formData,
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded'
      }
    })
      .then(response => {
        let imageURL = response.data.secure_url;
        return dispatch(addEvent(eventData, centerId, imageURL));
      })
      .catch(err => {
        dispatch({
          type: ADD_IMG_FAIL,
          error: 'Image upload failed'
        });
      });
  };
};
