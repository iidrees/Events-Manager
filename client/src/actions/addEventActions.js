/* Import dependencies and Modules */

import axios from 'axios';
import { history } from '../routes';

import { 
  ADD_EVENT, 
  ADD_EVENT_FAIL, 
  ADD_IMG_FAIL,
  GET_EVENT
} from './types';


/**
 * Axios will help make POST request to add event
 * @export {function} addEvent function action
 * @param {eventData} eventData object sent to the server
 * @param {index}  index {centerId}
 * @param {imgUrl} imgUrl {the cloudinary url}
 * @returns {JSON} JSON data containing events 
 */
export const addEvent = (eventData, index, imgUrl ) => {
  
  return (dispatch) => {
    return axios({
      method: 'post',
      url: `/api/v1/events/${index}`,
      data: { 
        title: eventData.title,
        date: eventData.date,
        time: eventData.time,
        center: eventData.center,
        description: eventData.description,
        type: eventData.type,
        imgUrl: imgUrl 
      },
      headers: {
        'x-access-token': localStorage.getItem('x-access-token')
      },
      withCredentials: true
    })
    .then((response) => {
      dispatch({ type: GET_EVENT, event: response.data.data})
    })
    .catch((err) => {
      dispatch({ type: ADD_EVENT_FAIL, error: err.response.data });
    })
  }
}


/**
 * @export {function} imageUpload function
 * @param {eventData} eventData object sent to the server
 * @param {index}  index {centerId}
 * @returns {URL} URL link returned is used as a parameter
 */
export const imageUpload = (eventData,index) => {
  
  let formData = new FormData();
  
  formData.append('file', eventData.imgFile)
  formData.append('upload_preset', process.env.UPLOAD_PRESET)
  return (dispatch) => {
    axios({
      method: 'post',
      url: process.env.CLOUDINARY_URL,
      data: formData,
      headers: {
        'Content-Type':'application/x-www-form-urlencoded'
      }
    })
    .then(response => {
      let imageURL = response.data.secure_url;
      console.log('cloudinary is done here>>>>>>')
      return dispatch(addEvent(eventData,index,imageURL))
    })
    .catch(err => {

      dispatch({
        type: ADD_IMG_FAIL, 
        error: 'Image upload failed'
      })
    })
  }
} 