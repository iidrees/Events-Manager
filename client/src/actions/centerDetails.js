import axios from 'axios';
import { history } from '../routes';

import {
  GET_CENTER,
  GET_CENTER_FAIL,
  CANCEL_EVENT,
  CANCEL_EVENT_FAIL
} from './types';

/**
 *  @param {centerId} centerId the center's index
 * @param {page} page - page number
 * @returns {JSON} object
 */
export const centerDetails = (centerId, page) => {
  return dispatch => {
    return axios({
      method: 'GET',
      url: `/api/v1/centers/${centerId}?page=${page}`,
      headers: {
        'x-access-token': localStorage.getItem('x-access-token')
      },
      withCredentials: true
    })
      .then(response => {
        dispatch({ type: GET_CENTER, response: response.data });
      })
      .catch(err => {
        dispatch({ type: GET_CENTER_FAIL, error: err.response.data });
      });
  };
};

/**
 * @param {index} eventId - eventId
 * @returns {void}
 */
export const cancelEvent = eventId => {
  return dispatch => {
    return axios({
      method: 'PUT',
      url: `/api/v1/cancelEvent/${eventId}`,
      headers: {
        'x-access-token': localStorage.getItem('x-access-token')
      },
      withCredentials: true
    })
      .then(response => {
        dispatch({ type: CANCEL_EVENT, response: response.data });
      })
      .catch(err => {
        dispatch({ type: CANCEL_EVENT_FAIL, error: err.response.data });
      });
  };
};
