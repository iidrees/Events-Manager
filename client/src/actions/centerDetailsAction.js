import axios from 'axios';
import { history } from '../routes';

import { 
  GET_CENTER,
  GET_CENTER_FAIL

  } from './types';

/**
 * 
 * 
 * @param {any} index the center's index
 * @returns {JSON} object
 */
export const centerDetails = (index,page) => { // eslint-disable-line
  return (dispatch) => {
   return axios({
      method: 'GET',
      url: `/api/v1/centers/${index}?page=${page}`,
      headers: {
        'x-access-token': localStorage.getItem('x-access-token')
      },
      withCredentials: true
    })
    .then((response) => {
      console.log('the details actions>>>', response.data)
      dispatch({ type: GET_CENTER, response: response.data })
    })
    .catch((err) => {
      dispatch({ type: GET_CENTER_FAIL, error: err.response.data});
    })
  }
}