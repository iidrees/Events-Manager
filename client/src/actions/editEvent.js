import axios from 'axios';
import { history } from '../routes';

import {
  EDIT_EVENT_LOAD,
  EDIT_EVENT,
  EDIT_EVENT_FAIL,
  ADD_IMG_FAIL
} from './types';

/**
 * Axios will help make PUT request to add event
 * @export {function}
 * @param {eventData} eventData - JSON
 * @param {eventId} eventId -
 * @param {any} imgUrl - the url for the image uploaded on cloudinary
 * @returns {JSON} JSON data containing events
 */
export const editEvent = (eventData, eventId, imgUrl) => {
  return dispatch => {
    return axios({
      method: 'PUT',
      url: `/api/v1/events/${eventId}`,
      data: {
        title: eventData.title,
        startDate: eventData.startDate,
        endDate: eventData.endDate,
        center: eventData.center,
        description: eventData.description,
        imgUrl: imgUrl
      },
      headers: {
        'x-access-token': localStorage.getItem('x-access-token')
      },
      withCredentials: true
    })
      .then(event => {
        dispatch({ type: EDIT_EVENT, event: event.data });
      })
      .catch(err => {
        dispatch({ type: EDIT_EVENT_FAIL, error: err.response.data });
      });
  };
};

/**
 * @export {function} imageUpload function
 * @param {eventData} eventData object sent to the server
 * @param {eventId}  eventId {eventId}
 * @returns {URL} URL link returned is used as a parameter
 */
export const imageUpload = (eventData, eventId) => {
  let formData = new FormData();

  formData.append('file', eventData.imgFile);
  formData.append('upload_preset', process.env.UPLOAD_PRESET);
  return dispatch => {
    dispatch({ type: EDIT_EVENT_LOAD });
    return axios({
      method: 'post',
      url: process.env.CLOUDINARY_URL,
      data: formData,
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded'
      }
    })
      .then(response => {
        let imageURL = response.data.secure_url;
        return dispatch(editEvent(eventData, eventId, imageURL));
      })
      .catch(err => {
        dispatch({
          type: ADD_IMG_FAIL,
          error: 'Image upload failed'
        });
      });
  };
};
