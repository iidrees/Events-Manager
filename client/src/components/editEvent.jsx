import React from 'react';
import { connect } from 'react-redux';
import { Link, Redirect, withRouter } from 'react-router-dom';
import axios from 'axios';
import decode from 'jwt-decode';

import NavBarMain from './NavBarMain.jsx';
import Footer from './footer.jsx';
import { editEvent } from '../actions/editEventAction';
import getCenters from '../actions/getCentersAction';
import { detailEvent } from '../actions/eventAction';
import { history } from '../routes';

/**
 * @param {event} event - Object
 * @class Addevents
 * @extends { React.Component }
 */
class EditEvent extends React.Component {
  /**
   * Creates an instance of EditEvent.
   * @param {any} props -
   * @memberof EditEvent
   */
  constructor(props) {
    super(props);
    this.state = {
      eventData: {
        ...this.props.event
      }
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  /**
   * @returns {JSON} Object
   * @memberof EditEvent
   */
  componentDidMount() {
    const { dispatch } = this.props;
    return dispatch(getCenters());
  }

  onChange = event => {
    this.setState({
      eventData: {
        ...this.state.eventData,
        [event.target.name]: event.target.value
      }
    });
  };

  onSubmit = event => {
    event.preventDefault();
    let { eventData } = this.state;
    const { dispatch } = this.props;
    return dispatch(editEvent(eventData, this.props.match.params.id));
  };

  /**
   * @returns {any} -
   * @memberof Addevents
   */
  render() {
    const { centers, event, user, editEvents } = this.props;

    return (
      <div>
        <div className="container">
          <div className="row">
            <div className="container" id="add-event-header">
              <div className="row">
                <div className="col-sm-12">
                  <h1 className="text-center head-1">Edit your events </h1>
                  <p className="head-para text-center">
                    Are you in need of a location to host your event? If yes,
                    then why not host your events using one of our numerous
                    event centers by creating an event below?
                  </p>
                </div>
              </div>
            </div>
            {editEvents.status === 'Success' && (
              <div className="alert alert-success" role="alert">
                <button
                  type="button"
                  className="close"
                  data-dismiss="alert"
                  aria-label="Close"
                >
                  <span aria-hidden="true">&times;</span>
                </button>
                <strong>{editEvents.message}.</strong>
              </div>
            )}
            {editEvents.status === 'Unsuccessful' && (
              <div className="alert alert-danger" role="alert">
                <button
                  type="button"
                  className="close"
                  data-dismiss="alert"
                  aria-label="Close"
                >
                  <span aria-hidden="true">&times;</span>
                </button>
                <strong>{editEvents.message}</strong>
                <span> </span>
                <strong>{editEvents.error}.</strong>
              </div>
            )}

            <div className="container">
              {/* <!-- Start container for Add Form --> */}
              <div className="row">
                <div className="col-sm-12">
                  <form
                    method="POST"
                    className="form-horizontal"
                    onSubmit={this.onSubmit}
                    role="form"
                    action="#"
                  >
                    {/* <!-- ADD FORM --> */}
                    <div className="form-group">
                      <label htmlFor="add-event" className=" home-para">
                        Name of Event:
                      </label>
                      <input
                        className="form-control"
                        onChange={this.onChange}
                        name="title"
                        id="form-event1"
                        type="text"
                        value={this.state.eventData.title}
                      />
                    </div>
                    <div className="form-group">
                      <label htmlFor="add-event" className=" home-para">
                        Date of event:
                      </label>
                      <input
                        className="form-control"
                        onChange={this.onChange}
                        name="date"
                        type="text"
                        id="form-event2"
                        placeholder="YY/MM/DD"
                        value={this.state.eventData.date}
                      />
                    </div>
                    <div className="form-group">
                      <label htmlFor="add-event" className=" home-para">
                        Time:
                      </label>
                      <input
                        className="form-control"
                        onChange={this.onChange}
                        name="time"
                        type="text"
                        id="form-event7"
                        placeholder="hh:mm AM/PM"
                        value={this.state.eventData.time}
                      />
                    </div>
                    <div className="form-group">
                      <label htmlFor="event-centers" className=" home-para">
                        Center:
                      </label>
                      <select
                        className="form-control"
                        id="event-center1"
                        onChange={this.onChange}
                        name="center"
                      >
                        <option
                          className="home-para"
                          key={this.state.eventData.centerId}
                        >
                          {this.state.eventData.center}
                        </option>
                        {centers.centers.map(center => {
                          return (
                            <option key={center.id} className="home-para">
                              {center.name}
                            </option>
                          );
                        })}
                      </select>
                    </div>
                    <div className="form-group">
                      <label htmlFor="event-details" className=" home-para">
                        Events description:
                      </label>
                      <textarea
                        className="form-control"
                        onChange={this.onChange}
                        id="form-event4"
                        rows="8"
                        name="description"
                        placeholder="brief details about the event"
                        value={this.state.eventData.description}
                      />
                    </div>
                    <div className="form-group">
                      <label htmlFor="add-event" className=" home-para">
                        Upload an Image of your event below:
                      </label>
                      <input
                        type="file"
                        className="form-control-file"
                        onChange={this.onChange}
                        aria-describedby="fileHelp"
                      />
                    </div>

                    <button
                      type="submit"
                      className="btn btn-primary btn-sm"
                      id="save-event"
                    >
                      Save{' '}
                      <span>
                        <i className="fa fa-paper-plane" aria-hidden="true" />
                      </span>
                    </button>
                  </form>
                </div>
                {/* <!-- End Container ADD EVENTS FORM --> */}
              </div>
            </div>
          </div>
          {/* <!-- End container for the Head section --> */}
        </div>
        <Footer />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    dispatch: action => dispatch(action)
  };
};
const mapStateToProps = state => {
  return {
    addevents: state.eventReducer,
    user: state.userReducer,
    centers: state.centerReducer,
    event: state.detailsEventReducer,
    editEvents: state.editEventReducer
  };
};
export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(EditEvent)
);
