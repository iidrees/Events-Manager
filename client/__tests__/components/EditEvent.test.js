import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount, render } from 'enzyme';
import { MemoryRouter as Router, withRouter } from 'react-router-dom';
import sinon from 'sinon';

import { event } from '../actions/mocks/addEventMock';
import { editEvent, imageUpload } from '../../src/actions/editEvent';
import { EditEvent } from '../../src/components/EditEvent';

describe('EditEvent component', () => {
  beforeEach(() => {
    jest.spyOn(console, 'error');
    console.error.mockImplementation(() => {});
  });

  afterEach(() => {
    console.error.mockRestore();
  });
  const locations = [];
  const history = {
    push(location) {
      locations.push(location);
    }
  };
  let props = {
    authenticated: true,
    editEvents: {
      status: 'Success',
      message: 'message',
      isLoading: true
    },
    match: {
      params: {
        id: 1
      }
    },
    centers: [],
    user: {},
    event: {},
    token:
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdXRoZW50aWNhdGVkIjp0cnVlLCJpZCI6MSwiYWRtaW4iOnRydWUsInJvbGUiOiJzdXBlckFkbWluIiwiaXNTdXBlckFkbWluIjp0cnVlLCJuYW1lIjoiSWRyZWVzIiwiaWF0IjoxNTI3NTk5OTkzLCJleHAiOjE1Mjc2MzU5OTN9.UAzE5IOIN3RVO7E4JCOnwtzpp5vberbZn8jOflEf1JI',

    onChange: jest.fn(),
    onSubmit: jest.fn(),
    onImageChange: jest.fn(),
    dispatch: jest.fn()
  };
  let wrapper = shallow(<EditEvent {...props} />);
  it('should render without throwing an error', done => {
    expect(mount(<EditEvent {...props} />).length).toEqual(1);
    done();
  });

  it('should respond to full rendering', done => {
    props = {
      editEvents: {
        status: 'Unsuccessful'
      },
      centers: {
        centers: [
          {
            id: 1,
            title: 'The Amity party',
            time: '10:00 PM',
            date: '21-03-2019',
            type: 'public',
            center: 'The Multi-purpose',
            description:
              'This is going to be the weirdest parry on the month of March',
            imgUrl:
              'https://static.pexels.com/photos/169193/pexels-photo-169193.jpeg'
          },
          {
            id: 2,
            title: 'The Amity party',
            time: '10:00 PM',
            date: '21-03-2019',
            type: 'public',
            center: 'The Multi-purpose',
            description:
              'This is going to be the weirdest parry on the month of March',
            imgUrl:
              'https://static.pexels.com/photos/169193/pexels-photo-169193.jpeg'
          }
        ]
      },
      user: {},
      match: {
        params: {
          id: 1
        }
      },
      event: {
        status: '',
        title: 'The Amity party',
        time: '10:00 PM',
        date: '21-03-2019',
        type: 'public',
        center: 'The Multi-purpose',
        description:
          'This is going to be the weirdest parry on the month of March',
        imgUrl:
          'https://static.pexels.com/photos/169193/pexels-photo-169193.jpeg'
      },
      token:
        'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdXRoZW50aWNhdGVkIjp0cnVlLCJpZCI6MSwiYWRtaW4iOnRydWUsInJvbGUiOiJzdXBlckFkbWluIiwiaXNTdXBlckFkbWluIjp0cnVlLCJuYW1lIjoiSWRyZWVzIiwiaWF0IjoxNTI3NTk5OTkzLCJleHAiOjE1Mjc2MzU5OTN9.UAzE5IOIN3RVO7E4JCOnwtzpp5vberbZn8jOflEf1JI',

      onChange: jest.fn(),
      onSubmit: jest.fn(),
      onImageChange: jest.fn(),
      dispatch: jest.fn()
    };
    const eventData = {
      status: '',
      title: 'The Amity party',
      time: '10:00 PM',
      date: '21-03-2019',
      type: 'public',
      center: 'The Multi-purpose',
      description:
        'This is going to be the weirdest parry on the month of March',
      imgUrl: 'https://static.pexels.com/photos/169193/pexels-photo-169193.jpeg'
    };
    wrapper = mount(
      <Router>
        <EditEvent {...props} history={history} />
      </Router>
    );

    expect(wrapper.find('#form-event1').length).toEqual(1);
    expect(wrapper.setState(eventData));
    wrapper.find('form').simulate('submit');
    props.onImageChange({ preventDefault: () => {} });
    props.onSubmit({ preventDefault: () => {} });
    expect(props.onSubmit).toBeCalled();
    done();
  });

  it('should respond to submit and image upload clicks', done => {
    wrapper.find('#input-file').simulate('change', {
      target: {
        name: 'images',
        files: [{ data: 'lakjdlhasjhdljhajf', type: 'image/jpeg' }]
      }
    });

    expect(wrapper.state('images')).toEqual(undefined);
    done();
  });
  it('should respond to submit and image upload clicks', done => {
    wrapper.find('#input-file').simulate('change', {
      target: {
        name: 'images',
        files: [{ data: 'lakjdlhasjhdljhajf', type: 'image/jpg' }]
      }
    });

    expect(wrapper.state('images')).toEqual(undefined);
    done();
  });
});
