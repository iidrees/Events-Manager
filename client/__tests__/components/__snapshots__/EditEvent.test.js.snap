// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditEvent component should render without throwing an error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditEvent
    authenticated={true}
    centers={Array []}
    dispatch={
      [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
          Array [
            [Function],
          ],
        ],
      }
    }
    editEvents={
      Object {
        "isLoading": false,
        "message": "message",
        "status": "",
      }
    }
    event={Object {}}
    onChange={[MockFunction]}
    onImageChange={[MockFunction]}
    onSubmit={[MockFunction]}
    token="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdXRoZW50aWNhdGVkIjp0cnVlLCJpZCI6MSwiYWRtaW4iOnRydWUsInJvbGUiOiJzdXBlckFkbWluIiwiaXNTdXBlckFkbWluIjp0cnVlLCJuYW1lIjoiSWRyZWVzIiwiaWF0IjoxNTI3NTk5OTkzLCJleHAiOjE1Mjc2MzU5OTN9.UAzE5IOIN3RVO7E4JCOnwtzpp5vberbZn8jOflEf1JI"
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "push": true,
      "to": "/myevents",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "push": true,
        "to": "/myevents",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
